cli/cmd/org/rename.go

```diff
@@ -40,51 +39,95 @@ func RenameCmd(ch *cmdutil.Helper) *cobra.Command {
 				}
 			}
 
-			if ch.Interactive {
-				err = cmdutil.SetFlagsByInputPrompts(*cmd, "new-name")
-				if err != nil {
-					return err
+			resp, err := client.GetOrganization(ctx, &adminv1.GetOrganizationRequest{Name: name})
+			if err != nil {
+				if st, ok := status.FromError(err); ok {
+					if st.Code() != codes.NotFound {
+						return err
+					}
 				}
+				fmt.Printf("Org name %q doesn't exist, please run `rill org list` to list available orgs\n", name)
+				return nil
+			}
+
+			org := resp.Organization
+			req := &adminv1.UpdateOrganizationRequest{
+				Name: org.Name,
 			}
 
-			if newName == "" {
-				return fmt.Errorf("please provide valid org new-name, provided: %q", newName)
+			ch.PrintfWarn("\nWarn: Renaming an org would invalidate dashboard URLs.\n")
+			hasDisplayName := org.DisplayName != ""
+
+			if cmd.Flags().Changed("display-name") {
+				req.DisplayName = &displayName
+			} else if ch.Interactive && !force && hasDisplayName {
+				displayName, err = cmdutil.InputPrompt("Enter the display name", org.DisplayName)
+				if err != nil {
+					return err
+				}
+				req.DisplayName = &displayName
 			}
 
-			if !force {
-				msg := fmt.Sprintf("Do you want to rename org \"%s\" to \"%s\"?", color.YellowString(name), color.YellowString(newName)) // nolint:gocritic // Because it uses colors
-				ok, err := cmdutil.ConfirmPrompt(msg, "", false)
+			if cmd.Flags().Changed("new-name") {
+				req.NewName = &newName
+				if !cmd.Flags().Changed("display-name") && org.DisplayName != "" {
+					ch.PrintfWarn("Warn: Changing org name without updating display name. Consider using --display-name to keep names consistent.\n")
+				}
+			} else if ch.Interactive && !force {
+				ok, err := confirmFieldChange("Org name", force)
```

Comment:
The case handling here is quite difficult to follow, but it seems like running `rill org rename NAME --display-name DISPLAY_NAME` will lead to a prompt to change the slug name? That would be dangerous because a non-breaking change could easily lead to accidentally breaking URLs. It would also violate the idea that we avoid prompts for optional fields when flags are provided. 

Overall, do you think you can simplify the various cases here? If it makes it simpler, it's okay to prompt less and/or have more errors.